File: 3 курс/AMO/S½ÑF¬á ña«ínº¬¿/ÇîÄ2014-15/èé-23/ÅÑGa¿tp¬/ïáí5/Trapezium.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab5
{
    class Trapezium 
    {
        double LOWER_LIMIT;
        double UPPER_LIMIT;
        double step;
        public Trapezium(double low, double up, double getstep)
        {
            LOWER_LIMIT = low;
            UPPER_LIMIT = up;
            step = getstep;
        }
       public double Solve (Func<double,double> integrated)  
        {
	        double iter = LOWER_LIMIT;
	        double end = UPPER_LIMIT - step;
	        double res = integrated(iter)/2;
            for (; iter <= end; iter += step)
                res += integrated(iter);
            res = step * ( integrated(iter)/ 2 + res);
	        return res;
        }
    }
}

File: 3 курс/AMO/S½ÑF¬á ña«ínº¬¿/ÇîÄ2014-15/èé-23/ï¿¼á¡/½áíá4/Trapezium.cpp
#include "Trapezium.h"
#include "const.h"


double Trapezium::Exact ()
{
	return eq.AntiDerivative(UPPER_LIMIT) - eq.AntiDerivative(LOWER_LIMIT); 
}
double Trapezium::Solve (double step)
{
	double iter = LOWER_LIMIT;
	double end = UPPER_LIMIT - step;
	double res = eq.Integrand(iter)/2;
	for ( ; iter <= end; iter+=step) 
		res += eq.Integrand(iter); 
	res = step*(eq.Integrand(iter)/2+res);
	res+= Rest((UPPER_LIMIT+LOWER_LIMIT)/2, step);
	return res;
}
double Trapezium::Rest (double x, double step)
{
	return -(UPPER_LIMIT-LOWER_LIMIT)*step*step/12*eq.SecondDerivative(x);
}